AFTests/HftTests/HftBaseTest.cs

```diff
@@ -17,23 +18,98 @@ class HftBaseTest : BaseTest
         protected string FirstAssetId = HFTSettings.GetHFTSettings().FirstAssetId;
         protected string SecondAssetId = HFTSettings.GetHFTSettings().SecondAssetId;
         protected string AssetPair = HFTSettings.GetHFTSettings().AssetPair;
+
+        // Market order
+        protected IResponse<MarketOrderResponseModel> CreateAndValidateMarketOrder(
+            string assetId, 
+            string assetPairId, 
+            OrderAction orderAction, 
+            double volume, 
+            string apiKey,
+            HttpStatusCode statusCode = HttpStatusCode.OK)
+        {
+            var request = new PlaceMarketOrderModel()
+            {
+                Asset = assetId,
+                AssetPairId = assetPairId,
+                OrderAction = orderAction,
+                Volume = volume
+            };
+
+            return hft.Orders.PostOrdersMarket(request, apiKey)
+                .Validate
+                .StatusCode(statusCode);
+        }
+
+        // Limit order
+        protected IResponse<LimitOrderResponseModel> CreateAndValidateLimitOrder(
+            double price,
+            string assetPairId,
+            OrderAction orderAction,
+            double volume,
+            string apiKey,
+            HttpStatusCode statusCode = HttpStatusCode.OK)
+        {
+            var request = new PlaceLimitOrderModel()
+            {
+                Price = price,
+                AssetPairId = assetPairId,
+                OrderAction = orderAction,
+                Volume = volume
+            };
+
+            return hft.Orders.PostOrdersLimitOrder(request, apiKey)
+                .Validate
+                .StatusCode(statusCode);
+        }
+
+        // Stop limit order    
+        protected IResponse<LimitOrderResponseModel> CreateAndValidateStopLimitOrder(
+            string assetPairId,
+            OrderAction orderAction,
+            double volume,
+            double lowerLimitPrice,
+            double lowerPrice,
+            double upperLimitPrice,
+            double upperPrice,
+            string apiKey,
+            HttpStatusCode statusCode = HttpStatusCode.OK)
+        {
+            var request = new PlaceStopLimitOrderModel()
+            {
+                AssetPairId = AssetPair,
+                OrderAction = orderAction,
+                Volume = volume,
+                LowerLimitPrice = lowerLimitPrice,
+                LowerPrice = lowerPrice,
+                UpperLimitPrice = upperLimitPrice,
+                UpperPrice = upperPrice
+            };
+
+            return hft.Orders.PostOrdersStopLimitOrder(request, apiKey)
+                .Validate
+                .StatusCode(statusCode);
```

Comment:
a little confused here
I see that the response is StatusCode, but the method declared as IResponse <T>
In fact, I think that even “Validate” method should not present here. Object, that describe http response should not validate itself. It is not its task

Reply:
Anyway, we can leave it here, but in my mind, should not be used in this way in future

Reply:
It's the short version of this:
"var response = hft.Orders.PostOrdersLimitOrder(request, apiKey);
response.Validate.StatusCode(statusCode);
return response;"
Because the method is "CreateAndValidate" and in this way, we skip writing additional Asserts for the status each time when we create orders. As you know the Validate method checks if the actual status code is the same with the expected and this method works for positive and negative tests.

Reply:
@vedat-murat thanks, that is clear now. But i still want to make all assertions in tests, not in other places, And other idea is to keep tests easy to read and understand. Some stuff like "return StatusCode" and mark method as IResponse<Type> give me some questions at the first sign.
Seems, it is ok for PR as the logic is clear at this moment